### PR TITLE
[Tune] Remove trial table when running Ray Train in a Jupyter notebook

### LINKED ISSUE #44776: [train] Remove trial table when running Ray Train in a Jupyter notebook
Ray Train only spawns a single "trial" with multiple distributed training workers. However, on Jupyter notebooks, Ray Train will still show a Ray Tune trial status table, which is only really useful when running multiple trials on Ray Tune. The fact that this shows up is an implementation detail of Ray Train being run on top of Ray Tune. We should remove it so that users are not confused about "Ray Tune status" when they never called any Ray Tune code themselves.

Screenshot 2024-04-16 at 10 58 13 AM

See https://github.com/ray-project/ray/blob/master/python/ray/tune/progress_reporter.py#L1525

### PR DESCRIPTION
**Pull request by hongpeng-guo**
## Why are these changes needed?

This PR removes the trial table mistakenly displayed during Ray Train in a Jupyter notebook, which should only appear in Ray Tune sessions.

image

## Related issue number

Closes #44776

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/tune/progress_reporter.py [hunk 1]
```diff
@@ -1520,7 +1520,8 @@ def _detect_reporter(**kwargs) -> TuneReporterBase:
 
     Keyword arguments are passed on to the reporter class.
     """
-    if IS_NOTEBOOK:
+    _trainer_api = kwargs.pop("_trainer_api", False)
```
Comment (justinvyu) [inline python/ray/tune/progress_reporter.py:1523]: We can add `_trainer_api` as a keyword argument separate from kwargs.

File: python/ray/tune/tests/test_progress_reporter.py [hunk 2]
```diff
@@ -801,6 +801,9 @@ def testReporterDetection(self):
             reporter = _detect_reporter()
             self.assertFalse(isinstance(reporter, CLIReporter))
             self.assertTrue(isinstance(reporter, JupyterNotebookReporter))
+            trainer_reporter = _detect_reporter(_trainer_api=True)
+            self.assertFalse(isinstance(trainer_reporter, JupyterNotebookReporter))
+            self.assertTrue(isinstance(trainer_reporter, CLIReporter))
```
Comment (justinvyu) [inline python/ray/tune/tests/test_progress_reporter.py:806]: Nice test!

**Review by justinvyu — COMMENTED** [review]
Just one small comment:

File: python/ray/tune/progress_reporter.py (hunk 1 above)
Comment (hongpeng-guo) [reply python/ray/tune/progress_reporter.py]: Just added a keyword arg `_trainer_api` for the function `_detect_reporter()`.

**Review by justinvyu — APPROVED** [review]
Thanks!

**Pull request merged**